src/okta/services/okta.service.ts

```diff
@@ -10,226 +10,82 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { Inject, Injectable, Injector } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
 import {
   assertIssuer,
   assertClientId,
   assertRedirectUri,
 } from '@okta/configuration-validation';
 
-import { OKTA_CONFIG, OktaConfig, AuthRequiredFunction } from '../models/okta.config';
-import { UserClaims } from '../models/user-claims';
-import { TokenManager, AccessToken, IDToken } from '../models/token-manager';
+import { OKTA_CONFIG, OktaConfig } from '../models/okta.config';
 
 // eslint-disable-next-line node/no-unpublished-import
 import packageInfo from '../packageInfo';
 
 /**
  * Import the okta-auth-js library
  */
-import OktaAuth from '@okta/okta-auth-js';
+import { OktaAuth, AuthState, OktaAuthOptions } from '@okta/okta-auth-js';
 import { Observable, Observer } from 'rxjs';
 
 @Injectable()
-export class OktaAuthService {
-    private oktaAuth: OktaAuth;
+export class OktaAuthService extends OktaAuth {
     private config: OktaConfig;
     private observers: Observer<boolean>[];
-    $authenticationState: Observable<boolean>;
 
-    constructor(@Inject(OKTA_CONFIG) config: OktaConfig, private injector: Injector) {
-      this.observers = [];
+    $authenticationState: Observable<boolean>;
 
-      /**
-       * Cache the auth config.
-       */
-      this.config = Object.assign({}, config);
-      this.config.scopes = this.config.scopes || ['openid', 'email'];
+    constructor(@Inject(OKTA_CONFIG) config: OktaConfig) {
+      // Assert Configuration
+      assertIssuer(config.issuer, config.testing);
+      assertClientId(config.clientId);
+      assertRedirectUri(config.redirectUri);
+
+      const transformAuthState = async (oktaAuth: OktaAuth, authState: AuthState) => {
+        // if `isAuthenticated` was set on config, call it now to override the value of `authState.isAuthenticated`
+        if (config.isAuthenticated) {
+          authState.isAuthenticated = await config.isAuthenticated(this);
+        }
+        // if `transformAuthState` was set on config, call it now to transform the authState object
+        if (config.transformAuthState) {
+          authState = await config.transformAuthState(oktaAuth, authState);
+        }
+        return authState;
+      };
+      
+      const options: OktaAuthOptions = Object.assign({
+        transformAuthState
+      }, config);
 
-      /**
-       * Scrub scopes to ensure 'openid' is included
-       */
+      super(options);
 
-      this.scrubScopes(this.config.scopes);
+      this.config = config;
```

Comment:
Looks like config is purely user input, use `this.options` instead?

Reply:
yes that's right, this.config is user input, this.options is the object used by okta-auth-js. There is a difference, as in the case where we are overriding a method (transformAuthState) but also allowing the user to set it.